### ISSUE TITLE
pointer as a default parameter bug

### ISSUE THREAD

**Issue by reluctant-git-user** [odin-lang/Odin#2869]
## Context

```
xyz:^int=transmute(^int)uintptr(1)

bug::proc(p:^int=xyz) { // pointer type argument with a default value
  fmt.println(p)
}

main::proc() {
  bug(nil) // does not pass 'nil', passes 'xyz' instead (the program prints '0x1')
}
```

* Operating System & Odin Version:
        Odin: dev-2023-10-nightly:2cca0050
        OS:   Windows 11 Home Basic (version: 22H2), build 22621.2428
        CPU:  AMD Ryzen 5 5600X 6-Core Processor
        RAM:  16310 MiB

## Expected Behavior

'nil' would be passed to the procedure which would print 0x0

## Current Behavior

default parameter is incorrectly passed and the program prints 0x1

### Steps to Reproduce

compile and run the sample program

**Comment by Kelimion** [comment]
To clarify, `bug(p=nil)` does do what you expect `bug(nil)` to do, correct?

**Comment by reluctant-git-user** [comment]
> To clarify, `bug(p=nil)` does do what you expect `bug(nil)` to do, correct?

No. That also passes the default argument instead of nil. If you remove the default (i.e. bug::proc(p:^int) {...) then the result is what I expect (program prints 0x0)

**Comment by Feoramund** [comment]
I cannot replicate this as of the latest commit. I get `0` when passing `nil` and `1` when nothing is passed.

**Comment by Feoramund** [comment]
Circling back to this: This is an interesting case, and as far as I understand the language semantics, it probably should work without error. I recently re-read parts of the overview and saw that default arguments must be known at compile-time, which this is, technically, despite referencing a variable and not a constant.
